azure-ts-webserver-component/index.ts

```diff
@@ -0,0 +1,21 @@
+import * as pulumi from "@pulumi/pulumi";
+import { WebServer } from "./webserver";
+
+// Get the desired username and password for our webserver VMs.
+let config = new pulumi.Config();
+let count = config.getNumber("count") || 2;
+let username = config.require("username");
+let password = config.requireSecret("password");
+
+// Now, allocate a few websever VMs -- by default, just 2, but this is configurable.
+export let ipAddresses = [];
+for (let i = 0; i < count; i++) {
+    let server = new WebServer(`ws-${i}`, {
+        username,
+        password,
+        bootScript: `#!/bin/bash\n
+echo "Hello, from Server #{i+1}!" > index.html
+nohup python -m SimpleHTTPServer 80 &`,
+    });
+    ipAddresses.push(server.getIpAddress());
```

Comment:
`map` would certainly read nicer than a mutable array. It might be a little foreign, but then again, that's one of the benefits of using JavaScript here. @lukehoban @CyrusNajmabadi Any opinion on this?

Reply:
i thnk the for-loop reads quite well.  So i wouldn't over-index here.  I find with heavyweight stuff (like making resources) sometimes being a bit more imperative/explicit feels a little better (to me at least).

Reply:
I agree that `map` is more natural for pure functions, not ones creating cloud resources. But at the same time I keep writing snippets like this https://github.com/pulumi/pulumi-gcp/issues/158#issuecomment-498644482 and I would hate to change it to a for-loop. Might be a personal pref.

Reply:
We will leave this as is for now!